backend/layers/processing/rollback.py

```diff
@@ -0,0 +1,301 @@
+        """
+        filter = CollectionQueryFilter(is_published=False)
+        collections = self.business_logic.get_collections(filter)
+        rolled_back_datasets = []
+        for collection in collections:
+            collection_datasets = self.rollback_private_collection(collection.version_id)
+            rolled_back_datasets.extend(collection_datasets)
+        self._clean_up_rolled_back_datasets(rolled_back_datasets)
+
+    def rollback_private_collection_list(self) -> None:
+        """
+        Rollback all the datasets from the input list of private collections. This will restore the
+        state of the list of private collections to their pre-migration state. Then, triggers deletion of the DB
+        references and S3 assets for the rolled back DatasetVersions.
+        """
+        rolled_back_datasets = []
+        for collection_version_id in self.entity_id_list:
+            try:
+                collection_dataset_versions = self.rollback_private_collection(
+                    CollectionVersionId(collection_version_id)
+                )
+            except Exception as e:
+                logger.info(f"Failed to rollback CollectionVersion {collection_version_id}, {e}")
+                continue
+            rolled_back_datasets.extend(collection_dataset_versions)
+        self._clean_up_rolled_back_datasets(rolled_back_datasets)
+
+    def rollback_private_collection(self, collection_version_id: CollectionVersionId) -> List[DatasetVersion]:
+        """
+        Rolls back the dataset versions for all datasets in the given private collection. This will restore the state of
+        the private collection to its pre-migration state.
+        """
+        collection_version = self.business_logic.get_collection_version(collection_version_id)
+        if collection_version is None:
+            raise CollectionNotFoundException(f"CollectionVersion {collection_version_id} not found")
+        for dataset in collection_version.datasets:
+            try:
+                self.rollback_private_dataset(dataset.version_id, collection_version_id)
+            except Exception as e:
+                logger.info(f"Failed to rollback DatasetVersion {dataset.version_id}, {e}")
+                continue
+        return collection_version.datasets
+
+    def rollback_private_dataset_list(self) -> None:
+        """
+        Rolls back the Datasets in the CollectionVersions associated with each DatasetVersionId passed in, to their
+        respective previous, most recently created DatasetVersion. Then, triggers deletion of the DB references and S3
+        assets for the rolled back DatasetVersions.
+        """
+        rolled_back_datasets = []
+        for dataset_version_id in self.entity_id_list:
+            dataset_version_id = DatasetVersionId(dataset_version_id)
+            try:
+                rolled_back_dataset = self.rollback_private_dataset(dataset_version_id)
```

Comment:
I'm finding it confusing to distingush between `rollback_{public,private}_{collection, dataset}_list`. For readability consider reodering the function name to include the entity first ex: `private_dataset_rollback`, `private_dataset_list_rollback`. Alternatively `dataset_private_rollback`, `dataset_private_list_rollback`.